compiler/rustc_expand/src/proc_macro_server.rs

```diff
@@ -445,7 +451,8 @@ impl server::TokenStreamIter for Rustc<'_> {
         loop {
             let tree = iter.stack.pop().or_else(|| {
                 let next = iter.cursor.next_with_joint()?;
-                Some(TokenTree::from_internal((next, self.sess, &mut iter.stack)))
+                let lookahead = iter.cursor.look_ahead(0);
```

Comment:
Could you make this `look_ahead` return `Option<&TokenTree>` (by reference)?
Cloning a token tree is relatively cheap, but it still potentially involves an atomic increment and it's not something I'd want to be done for every iterated token.

Reply:
Fixed!